test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1828 @@
+        self.misc_info = {}
+        self.anon_obj = None
+        self.all_hosts = []
+        self.server = None
+        self.scheduler = None
+        self.log_utils = PBSLogUtils()
+        self.outtar_path = None
+        self.outtar_fd = None
+        self.create_tar = create_tar
+        self.snapshot_name = None
+        self.sudo = sudo
+        self.log_path = log_path
+        if self.log_path is not None:
+            self.log_filename = ntpath.basename(self.log_path)
+        else:
+            self.log_filename = None
+
+        # finalize() is called by the context's __exit__() automatically
+        # however, finalize() is non-reenterant, so set a flag to keep
+        # track of whether it has been called or not.
+        self.finalized = False
+
+        # Parse the input arguments
+        # Uses Python's ternary conditional construct:
+        #    a if <condition> else b
+        self.diag = diag_path
+        self.outdir = os.path.abspath(out_dir)
+        self.snapshot_name = ntpath.basename(self.outdir)
+        self.server_host = server_host
+        self.num_acct_logs = int(acct_logs) if acct_logs is not None else 0
+        self.num_service_logs = int(srvc_logs) if srvc_logs is not None else 0
+        self.all_hosts = []
+        if additional_hosts is not None:
+            hosts_str = "".join(additional_hosts.split())  # remove whitespaces
+            self.all_hosts.extend(hosts_str.split(","))
+        self.mapfile = map_file
+        self.anonymize = anonymize
+        if self.anonymize is True:
+            # Create a PBSAnonymizer object
+            del_attrs = [ATTR_v, ATTR_mailfrom, ATTR_managers, ATTR_owner,
+                         ATTR_operators, ATTR_m, ATTR_aclgroup,
+                         ATTR_acluser, ATTR_aclResvgroup, ATTR_aclResvuser,
+                         ATTR_name, ATTR_jobdir, ATTR_submit_arguments]
+            obf_attrs = [ATTR_euser, ATTR_egroup, ATTR_project, ATTR_A,
+                         ATTR_g, ATTR_M, ATTR_u, ATTR_SvrHost]
+            self.anon_obj = PBSAnonymizer(attr_delete=del_attrs,
+                                          attr_val=obf_attrs)
+
+        # Initialize Server and Scheduler objects
+        self.server = self.__initialize_server(self.server_host, self.diag)
+        self.scheduler = Scheduler(server=self.server, diag=self.diag)
+
+        # Add self.server_host to the list of hosts
+        self.all_hosts.append(self.server_host)
+
+        # If output needs to be a tarball, create the tarfile name
+        # tarfile name = <output directory name>.tgz
+        self.outtar_path = self.outdir + ".tgz"
+
+        # Setup some infrastructure
+        self.__define_constants()
+        self.__set_paths_and_cmds()
+
+        # Create the snapshot directory tree
+        self.__initialize_snapshot()
+
+    def __define_constants(self):
+        """
+        Define some constants to use as indexes into the various dicts
+        """
+        self.QSTAT_B_OUT = 0
```

Comment:
You can make these variables as module variables and you can look at https://github.com/PBSPro/pbspro/blob/master/test/fw/ptl/lib/pbs_testlib.py#L103 to define them